### PR TITLE
Remove gpuCI scripts.

### PR DESCRIPTION
**Pull request by bdice**
Since migrating to GitHub Actions, the gpuCI scripts are no longer needed. This PR removes those outdated gpuCI scripts.

### CODE REVIEW

File: readme_pages/SOURCEBUILD.md [hunk 1]
```diff
@@ -233,11 +233,6 @@ Run either the C++ or the Python tests with datasets
 
 Note: This conda installation only applies to Linux and Python versions 3.8/3.10.
 
-### Building and Testing on a gpuCI image locally
```
Comment (ChuckHastings) [inline readme_pages/SOURCEBUILD.md:236]: Should we provide updated instructions on this?  With the new GA approach, there are new scripts that provide this behavior.  Is there documentation somewhere that describes how one would run the process that mimics CI that we could link to here?

**Review by ajschmidt8 — COMMENTED** [review]
I can't add a code suggestion for this, but we should also remove any `preserve_egg_dir: True` lines from the conda recipes.

These were added back with the `setuptools` pin in an attempt to resolve a Jenkins-image related conda issue.

It seems like it was copied and pasted to multiple new recipes.

File: readme_pages/SOURCEBUILD.md (hunk 1 above)
Comment (ajschmidt8) [reply readme_pages/SOURCEBUILD.md:236]: You're ahead of the curve, @ChuckHastings.

I still have to get that new functionality properly documented somewhere.

I'll try to get to it this week.

Comment (ChuckHastings) [reply readme_pages/SOURCEBUILD.md:236]: One of my life curses :-). Always a challenge when you live ahead of the curve.

**Review by ajschmidt8 — APPROVED** [review]

File: readme_pages/SOURCEBUILD.md (hunk 1 above)
Comment (ajschmidt8) [reply readme_pages/SOURCEBUILD.md:236]: Opened a PR for these new docs below.

- https://github.com/rapidsai/docs/pull/335

Comment (ajschmidt8) [reply readme_pages/SOURCEBUILD.md:236]: @ChuckHastings, with that new documentation almost published, would you mind approving this PR so we can get this merged?

**Review by rlratzel — APPROVED** [review]
LGTM.

This could be more of a suggestion for a separate PR, but can we add documentation somewhere on how to run the CI style checks locally, since the unofficial/undocumented suggestion before was to simply run the now-removed style.sh script?  I believe there may be docs somewhere on how to install the pre-commit hooks somewhere, but I can't find it, and having it in CONTRIBUTING or SOURCEBUILD or something would be great.

**Comment by bdice** [comment]
> LGTM.
> 
> This could be more of a suggestion for a separate PR, but can we add documentation somewhere on how to run the CI style checks locally, since the unofficial/undocumented suggestion before was to simply run the now-removed style.sh script? I believe there may be docs somewhere on how to install the pre-commit hooks somewhere, but I can't find it, and having it in CONTRIBUTING or SOURCEBUILD or something would be great.

@rlratzel Feel free to use this section of the cuDF contributing guide as a model: https://github.com/rapidsai/cudf/blob/branch-23.04/CONTRIBUTING.md#code-formatting

**Pull request merged**